src/rule_definition_tools.jl

```diff
@@ -1,6 +1,13 @@
 # These are some macros (and supporting functions) to make it easier to define rules.
 using MuladdMacro: @muladd
 
+"""
+    tail(x::Tuple)
+
+Produce a new `Tuple` containing all but the first element of `x`.
+"""
+tail(x::Tuple) = x[2:end]
```

Comment:
why not `Base.tail`?

Reply:
Indeed.
Also why have this in this file at all, given users can use `Base.tail` or `Base.peel`

Reply:
Both excellent points. The reason is that I didn't realise `Base.tail` existed.